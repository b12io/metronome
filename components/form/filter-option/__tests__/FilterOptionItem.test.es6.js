import React from 'react'
import renderer from 'react-test-renderer'
import FilterOptionItem from '../FilterOptionItem.es6.js'

describe('FilterOptionItem component', () => {
  let testProps

  beforeEach(() => {
    testProps = {
      filter: {
        field_name: 'id',
        operator: 'gte',
        value: 1
      },
      filterableFields: [],
      handleUpdateFilter: jest.fn()
    }
  })

  test('should not display content by default', () => {
    const tree = renderer.create(<FilterOptionItem {...testProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should display content', () => {
    const tree = renderer
      .create(<FilterOptionItem {...testProps} isOpened />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
