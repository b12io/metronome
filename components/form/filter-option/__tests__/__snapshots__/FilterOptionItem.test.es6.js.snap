// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterOptionItem component should display content 1`] = `
<div
  className="ds-filter-option__item"
>
  <div
    className="ds-filter-option__item-toggle is-active"
    onClick={[Function]}
  >
    Id
  </div>
  <div
    className="ds-filter-option__item-content"
  >
    <div
      className="ds-form-group"
    >
      <div
        className="ds-form-group ds-form-group--select"
      >
        <div
          className="ds-form-control-select"
        >
          <div
            className="ds-form-control ds-form-control-select__toggle"
            onClick={[Function]}
          >
            Id
          </div>
          <div
            className="ds-form-control-select__dropdown"
          >
            <div
              className="ds-form-control-select__dropdown-options"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ds-form-group"
    >
      <div
        className="ds-form-group ds-form-group--select"
      >
        <div
          className="ds-form-control-select"
        >
          <div
            className="ds-form-control ds-form-control-select__toggle"
            onClick={[Function]}
          >
            Please select
          </div>
          <div
            className="ds-form-control-select__dropdown"
          >
            <div
              className="ds-form-control-select__dropdown-options"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ds-form-group"
    >
      <div
        className="ds-form-group"
      >
        <div
          className="ds-form-control-wrap"
        >
          <input
            className="ds-form-control"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type="text"
            value={1}
          />
        </div>
      </div>
    </div>
    <div
      className="ds-form-group"
    >
      <button
        className="button button--danger button--sm button--block"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Remove filter
        
      </button>
    </div>
  </div>
</div>
`;

exports[`FilterOptionItem component should not display content by default 1`] = `
<div
  className="ds-filter-option__item"
>
  <div
    className="ds-filter-option__item-toggle"
    onClick={[Function]}
  >
    Id
  </div>
</div>
`;
